### PR TITLE
Docs/616 contributing

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Description

Update to contribution guidelines.

Issue/s resolved: #616 

## Changes proposed:
- Added intro with most important info and link to Mattermost
- Few edits in the standard guidelines 
- Added link to upcoming `documentation_howto.rst`, **this link currently points to the 504-docstring-formatting branch**, must be updated once that is merged.

## Type of change

- Documentation update

## Due Diligence

- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -43,7 +50,8 @@ git pull upstream master
 git checkout -b features/123-boolean-operators
 ```
 
-* Commit locally as you progress (`git add` and `git commit`) Use a properly formatted commit message, write tests that fail before your change and pass afterward, run all the tests locally and in parallel for different process counts (`mpirun -np <PROCESSES>`). Be sure to document any changed behavior in docstrings, keeping to HeAT's docstring standard (explained below).
+* Commit locally as you progress (`git add` and `git commit`) Use a properly formatted commit message, write tests that fail before your change and pass afterward, run all the tests locally and in parallel for different process counts (`mpirun -np <PROCESSES>`). Be sure to document any changed behavior in docstrings, keeping to HeAT's [docstring standard](https://github.com/helmholtz-analytics/heat/blob/504-docstring-formatting/doc/source/documentation_howto.rst).
```
Comment (Markus-Goetz) [inline contributing.md:53]: The link to the docstrings branch is somewhat unfortunate. Should have this as a TODO in 504 to let it point to master

File: contributing.md [hunk 2]
```diff
@@ -55,9 +63,9 @@ git push origin features/123-boolean-operators
 
 * Enter your GitHub username and password (advanced users can remove this step by connecting to GitHub with SSH.
 
-* Go to GitHub. The new branch will show up with a green Pull Request button. Make sure the title and message are clear, concise, and self- explanatory. Then click the button to submit it.
+* Go to GitHub. The new branch will show up with a green Pull Request button. Make sure the title and message are clear, concise, and self-explanatory. Then click the button to submit it.
```
Comment (Markus-Goetz) [inline contributing.md:66]: Perhaps a statement on rebasing here?

File: contributing.md (hunk 1 above)
Comment (ClaudiaComito) [reply contributing.md:53]: Agreed.

File: contributing.md (hunk 2 above)
Comment (ClaudiaComito) [reply contributing.md:66]: Hmm. Thanks for reminding me (to rebase).

Do we know what we want well enough to write it down in the guidelines? Do we want to ask the contributors to tidy up their commits only keeping the most important ones (`git rebase -i master` and then, in most cases, force push), or do we want to rebase by default before every PR? We haven't been sticking to any particular rule so far...

Comment (Markus-Goetz) [reply contributing.md:66]: They should rebase before the PR. After the PR is live, no more rebasing. Moreover, should we make a general reference with link towards our development process (what is it now btw Github Flow or Gitlab Flow?)

File: contributing.md [hunk 3]
```diff
@@ -1,6 +1,13 @@
 ## Contributing to HeAT
 
-*These contribution guidelines are inspired by NumPy's contribution guidelines of the SciPy community.*
+Thanks for your interest in contributing to HeAT!
```
Comment (Cdebus) [inline contributing.md:3]: "Thank you"....
And maybe add something: "We appreciate your effort" ?

File: contributing.md (hunk 1 above)
Comment (Cdebus) [inline contributing.md:53]: Dot missing after  (`git add` and `git commit`) ... as in end of sentence

Comment (Cdebus) [inline contributing.md:53]: I like the link to the docstrings... can we mention that again further down under "guidelines" ? (line 100 )

File: contributing.md (hunk 2 above)
Comment (ClaudiaComito) [reply contributing.md:66]: I don't think we have reached an agreement on this. I would leave it all out for now.

File: contributing.md (hunk 3 above)
Comment (ClaudiaComito) [reply contributing.md]: Done

File: contributing.md (hunk 1 above)
Comment (ClaudiaComito) [reply contributing.md]: Good catch! Done

Comment (ClaudiaComito) [reply contributing.md]: Yes good idea, done

File: contributing.md (hunk 2 above)
Comment (ClaudiaComito) [reply contributing.md:75]: I've mentioned rebasing before pushing as we discussed offline.

**Comment by ClaudiaComito** [comment]
@Markus-Goetz @Cdebus  I'm unhappy about the `pre-commit` information being on README. I'm going to move it to contributing.md and have README point to the contribution guidelines, unless somebody screams.

**Comment by Markus-Goetz** [comment]
Seems reasonable

**Comment by ClaudiaComito** [comment]
> Seems reasonable

I think I'm done. See you next month!

**Pull request merged**